### PR TITLE
OB - Bowling Scorecard

### PR DESCRIPTION
**Pull request by bateso88**
Still needs work. Would like to:
- Isolate unit tests for game
- Extract scorecard class from game
- Test against edge cases

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+require './lib/game'
+
+game = Game.new
```
Comment (GlykeriaStr) [inline app.rb:5]: I like the idea of the app so you won't have to run the game in stupid irb

File: README.md [hunk 2]
```diff
@@ -63,3 +39,11 @@ In the image below you can find some score examples.
 More about ten pin bowling here: http://en.wikipedia.org/wiki/Ten-pin_bowling
 
 ![Ten Pin Score Example](images/example_ten_pin_scoring.png)
+
+### To Complete:
+
+* Make it impossible for a user to score more than 10 in a frame (and other edge cases).
+* Isolate unit tests for game.
+* Extract scorecard class from game
```
Comment (GlykeriaStr) [inline README.md:47]: Great coding to have more than one class

File: lib/frame.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+# Frame class
+class Frame
+  STRIKE = 10
```
Comment (GlykeriaStr) [inline lib/frame.rb:5]: Constants! Smort!

File: lib/frame.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+  def tenth_frame_finished?
+    return false if roll_two.nil?
+
+    calculate_if_tenth_frame_finished_where_spare_or_strike
+  end
+
+  def calculate_if_tenth_frame_finished_where_spare_or_strike
+    if spare_or_strike?
+      if strike? && roll_two != STRIKE
```
Comment (GlykeriaStr) [inline lib/frame.rb:62]: You could have written this in one line so you can avoid the nested ifs?

File: lib/game.rb [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+      print_each(frame)
+    end
+    puts "Total: #{total_score}"
+  end
+
+  private
+
+  def print_each(frame)
+    if frame.number < TOTAL_NUMBER_OF_FRAMES
+      puts "Frame: #{frame.number} Roll 1: #{frame.roll_one} Roll 2: #{frame.roll_two} Score: #{frame.score}"
```
Comment (GlykeriaStr) [inline lib/game.rb:68]: It makes it look really nice that it prints your rolls and score after every roll!

File: spec/frame_spec.rb [hunk 6]
```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'frame'
+
+describe Frame do
```
Comment (GlykeriaStr) [inline spec/frame_spec.rb:5]: Amazing tests on this class!!

File: spec/game_spec.rb [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+      expect(game.current_frame_number).to eq 2
+    end
+    it 'returns two after a strike' do
+      game.bowl(10)
+      game.bowl(1)
+      expect(game.current_frame_number).to eq 2
+    end
+  end
+  describe '.print_scorecard' do
+    it 'should return nil' do
```
Comment (GlykeriaStr) [inline spec/game_spec.rb:27]: cheeky

**Review by GlykeriaStr — COMMENTED** [review]
Amazing work! Very impressed with the testing and the devising of classes! I also liked that you had a method for everything!
